examples/temperature-measurement-app/esp32/main/diagnostic-logs-provider-delegate-impl.cpp

```diff
@@ -29,6 +29,11 @@ using namespace chip::app::Clusters::DiagnosticLogs;
 LogProvider LogProvider::sInstance;
 LogProvider::CrashLogContext LogProvider::sCrashLogContext;
 
+#if CONFIG_ENABLE_ESP_DIAGNOSTICS_TRACE
```

Comment:
```suggestion
#ifdef CONFIG_ENABLE_ESP_DIAGNOSTICS_TRACE
```
else `-Wundef ` would throw error.

Please change the other usage of this config as well